src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,72 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * A class which encapsulates a task which has
+ * a deadline.
+ */
+public class Deadline extends Task {
+
+    /**
+     * The date representing the deadline of the task
+     */
+    protected LocalDate by;
+
+    /**
+     * A public constructor to initialise the deadline
+     * of the task and calls the parent constructor.
+     *
+     * @param description The description of the task.
+     * @param by The deadline of the task.
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Returns a string containing the given date in a
+     * different format. The format is changed to
+     * "MM d yyyy".
+     *
+     * @return The string representing the date.
+     */
+    public String changeDateFormat() {
+
```

Comment:
I'm not sure if this extra blank line here is intended? Same with the other few extra lines below.